src/sqlancer/GlobalState.java

```diff
@@ -118,6 +125,54 @@ public boolean executeStatement(Query q) throws SQLException {
         return success;
     }
 
+    public ResultSet executeStatementAndGet(Query q) throws SQLException {
+        boolean logExecutionTime = getOptions().logExecutionTime();
+        ExecutionTimer timer = logExecution(q, logExecutionTime);
+        ResultSet result = manager.executeAndGet(q);
+        if (result != null && getOptions().printSucceedingStatements()) {
+            System.out.println(q.getQueryString());
+        }
+        if (logExecutionTime) {
+            getLogger().writeCurrent(" -- " + timer.end().asString());
+        }
+        if (q.couldAffectSchema()) {
+            updateSchema();
+        }
+        return result;
+    }
+
+    public boolean fillAndExecuteStatement(Query q, String template, List<String> fills) throws SQLException {
```

Comment:
I think it would be cleaner to just extend the current `executeStatement` method. Rather than passing `List<String> fills`, you can specify `String... fills` as a signature, to specify variable-length arguments. This way, existing code will work without modification, and users of that method can optionally specify the additional fills when needed. Using varargs might also make it easier to use the method, since no intermediate list needs to be executed. For example, the `executeStatement` method could be called like this: `executeStatement(q, templateString, fill1, fill2, fill3)`.